### PR TITLE
chore: adapt arguments in MsgChannelUpgradeAck

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

Adapt the args supplied to ibc core in `MsgChannelUpgradeAck`

needs: #3702 
ref: #1620 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/core/04-channel/types/msgs.go [hunk 1]
```diff
@@ -637,17 +638,14 @@ func (msg MsgChannelUpgradeAck) ValidateBasic() error {
...
 	}
-	if msg.CounterpartyChannel.State != TRYUPGRADE {
-		return errorsmod.Wrapf(ErrInvalidChannelState, "expected: %s, got: %s", "TRYUPGRADE", msg.CounterpartyChannel.State)
-	}
 	if _, err := sdk.AccAddressFromBech32(msg.Signer); err != nil {
 		return errorsmod.Wrapf(ibcerrors.ErrInvalidAddress, "string could not be parsed as address: %v", err)
 	}
 
-	return nil
+	return msg.CounterpartyUpgrade.ValidateBasic()
```
[the remark this answers is not in the document]
Comment (charleenfei) [reply modules/core/04-channel/types/msgs.go:648]: i had some written in the UpgradeTimeout pr, if you want to copy from there: https://github.com/cosmos/ibc-go/pull/3535

[the remark this answers is not in the document]
Comment (damiannolan) [reply modules/core/04-channel/types/msgs.go:648]: Nice! I'll add you as co-author on merge 🙏

**Review by DimitrisJim — APPROVED** [review]
lgtm!

**Review by colin-axner — APPROVED** [review]

File: modules/core/04-channel/types/msgs.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (charleenfei) [reply modules/core/04-channel/types/msgs.go:648]: that's very kind <3

**Review by charleenfei — APPROVED** [review]
nice work, lgtm!

File: modules/core/04-channel/types/timeout_test.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+func (suite *TypesTestSuite) TestIsValid() {
+	var timeout types.Timeout
+
+	testCases := []struct {
+		name     string
+		malleate func()
+		expPass  bool
+	}{
+		{
```
Comment (chatton) [inline modules/core/04-channel/types/timeout_test.go:16]: should we add some test cases for negative timeout timestamp/height?

**Review by chatton — APPROVED** [review]
LGTM, left one small comment but nothing major 🥇

File: modules/core/04-channel/types/timeout_test.go (hunk 2 above)
Comment (chatton) [reply modules/core/04-channel/types/timeout_test.go:16]: oh then it should be no problem!
